pkg/plugin/v1/plugin.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+)
+
+type Plugin struct {
+	initPlugin
+	createAPIPlugin
+	createWebhookPlugin
+}
+
+var _ plugin.Base = Plugin{}
+
+func (_ Plugin) Name() string {
+	return "go"
+}
+
+func (_ Plugin) Version() string {
+	return "1"
```

Comment:
I think that may rise a lot of issues as we are expecting the default `go` plugin to exist for every version and using `Bar` as version would break that.

Reply:
Version formatting and plugin composition go hand-in-hand, so we should follow up this PR with a discussion on both.

Reply:
yeah.  I think for the moment: add a `TODO` for this, and we'll track them all down before final merge.

We can do the "move fast and break things" strategy on this branch.